### ISSUE TITLE
Persistent data not restored. 

### ISSUE THREAD

**Issue by dsphper** [mher/flower#751]
Environment
Python: 3.4
Celery: 4.1.0
Flower: 0.9.0
Centos: 7.0
--persistent flag is used. Celery version v4.1.0.

If I create a couple of tasks, they run as expected.

After I send a SIGINT:

[D 150923 14:43:09 events:96] Saving state to 'flower'...
[D 150923 14:43:09 events:97] <State: events=54 tasks=4>

The DB file 'flower' clearly contains the correct data. When I start flower again:

[D 150923 14:47:35 events:76] Loading state from 'flower'...
[D 150923 14:47:35 events:80] <State: events=0 tasks=0>

If I run Python and load the file with shelve:

> f['events']
<State: events=0 tasks=0>

So, something isn't working correctly when shelve reads the file.
Wait all day.
Please help me.

**Comment by ShaheedHaque** [comment]
Hi,

I'm running the following:

    flower   -> flower:0.9.2 tornado:4.5.3 babel:2.3.4
    software -> celery:4.1.0 (latentcall) kombu:4.1.0 py:3.6.3
                billiard:3.5.0.3 py-amqp:2.2.2
    platform -> system:Linux arch:64bit, ELF imp:CPython
    loader   -> celery.loaders.default.Loader
    settings -> transport:amqp results:disabled

and saving/restoring formt he --persistent database seems to work just fine (though this version does not emit the debug logs you saw). HOWEVER, the big difference is that I used SIGTERM based on the fact that the code - at least the version I am using, only establishes a signal handler for SIGTERM, not SIGINT. I did try SIGINT, and as expected, the Flower process died without updating the db.

I note that having scanned the docs, I don't see any reference to which signals are supported.
